tests/unit/externalRequestsTest.js

```diff
@@ -123,6 +495,118 @@ tap.test('External Requests Middleware', {autoend: true}, t => {
     )
   })
 
+  t.test('setKoaResponseBody()', {autoend: true}, t => {
+    t.test('should set the koa response', t => {
+      const ctx = {
+        body: {}
+      }
+      const body = {message: 'success'}
+      const request = {
+        id: '1233'
+      }
+
+      setKoaResponseBody(ctx, request, body)
+      t.deepEqual(ctx.body[request.id], body)
+      t.end()
+    })
+
+    t.test('should not set the koa response (primary response exists)', t => {
+      const ctx = {
+        body: {},
+        hasPrimaryRequest: true
+      }
+
+      const request = {
+        id: '1233'
+      }
+
+      setKoaResponseBody(ctx, request, null)
+      t.equals(ctx.body[request.id], undefined)
+      t.end()
+    })
+  })
+
+  t.test('setKoaResponseFromPrimary()', {autoend: true}, t => {
+    t.test('should set the koa response', t => {
+      const ctx = {
+        body: {}
+      }
+      const body = {message: 'success'}
+      const request = {
+        id: '1233'
+      }
+
+      setKoaResponseBodyFromPrimary(ctx, request, body)
+      t.deepEqual(ctx.body[request.id], body)
+      t.equals(ctx.hasPrimaryRequest, true)
+      t.end()
+    })
+  })
+
+  t.test('handleRequestError()', {autoend: true}, t => {
+    t.test('should return the error when there is a connection error', t => {
+      const ctx = {
+        body: {}
+      }
+      const request = {}
+      const err = {
+        message: 'ECONREFUSED'
```

Comment:
Im not sure the spelling of this error message is correct. I think it should be `ECONNREFUSED`

Reply:
done in https://github.com/jembi/openhim-mediator-mapping/pull/39/commits/f6e854686999d065f57f20f9f1af3f9e1318f21e